File: src/main/Java/boj_step2/boj2884.java
import java.io.*;
import java.util.StringTokenizer;

public class boj2884 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int h = Integer.parseInt(tokenizer.nextToken());
        int m = Integer.parseInt(tokenizer.nextToken());
        if (m >= 45) m -= 45;
        else {
            if (h == 0) h += 23;
            else h -=1;
            m += 15;
        }
        writer.append(h + " " + m);
        writer.close();
    }
}

File: src/main/kotlin/boj2884.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    var (h, m) = r.readLine()!!.split(" ").map { it.toInt() }
    if (m >= 45) m -= 45
    else {
        if (h == 0) h += 23
        else h -= 1
        m += 15
    }
    w.write(h.toString() + " " + m.toString())
    w.close()
}